fontforge/autosave.c

```diff
@@ -25,136 +25,121 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 #include "fontforgevw.h"
-/*#include "ustring.h"*/
-#include <sys/stat.h>
 #include <sys/types.h>
-#include <fcntl.h>
 #include <unistd.h>
-#include <errno.h>
-#include <stdlib.h>
-#include <dirent.h>
-#include <ustring.h>
-#include "gfile.h"
-#include "views.h"
-#include "gwidget.h"
 
 int AutoSaveFrequency=5;
 
-#if !defined(__MINGW32__)
-# include <pwd.h>
-#endif
-
-static char *getAutoDirName(char *buffer) {
-    char *dir=getFontForgeUserDir(Config);
-
-    if ( dir!=NULL ) {
-        sprintf(buffer,"%s/autosave", dir);
-        free(dir);
-        if ( access(buffer,F_OK)==-1 )
-            if ( GFileMkDir(buffer)==-1 )
-                return( NULL );
-        dir = copy(buffer);
+static char *GetAutoDirName() {
+    char *userdir = getFontForgeUserDir(Config);
+    char *autodir = NULL;
+    
+    if (userdir != NULL) {
+        int ret = asprintf(&autodir, "%s/autosave", userdir);
+        free(userdir);
+        if (ret == -1) {
+            return NULL;
+        } else if (!GFileExists(autodir) && GFileMkDir(autodir) == -1) {
```

Comment:
A comment explaining that we are creating the directory if non-existent might be in order.